client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -1529,35 +1529,24 @@ def _before_repre_load(self, placeholder, representation):
 
         pass
 
-    def _reduce_last_version_repre_entities(self, representations):
+    def _reduce_last_version_repre_entities(self, repre_contexts):
         """Reduce representations to last verison."""
 
         mapping = {}
-        # TODO use representation context with entities
-        # - using 'folder', 'subset' and 'version' from context on
-        #   representation is danger
-        for repre_entity in representations:
-            repre_context = repre_entity["context"]
+        for repre_context in repre_contexts:
+            folder_id = repre_context["folder"]["id"]
+            product_id = repre_context["product"]["id"]
+            version = repre_context["version"]["version"]
 
-            folder_name = repre_context["asset"]
-            product_name = repre_context["subset"]
-            version = repre_context.get("version", -1)
+            parents_path = "/".join([folder_id, product_id])
```

Comment:
Isn't the `product_id` ensured to be unique by itself? So that we could just reduce to the last version per product id?

Reply:
> Isn't the product_id ensured to be unique by itself? 

It is :) Will change.

> Also, if we are just using this as a key to the mapping can't we just use a tuple instead of concatenated string?

The path is in fact not used, but the versions mapping is used as it's sorting keys for latest only.